src/iop/denoiseprofile.c

```diff
@@ -2934,30 +2953,101 @@ void cleanup_global(dt_iop_module_so_t *self)
   self->data = NULL;
 }
 
-static dt_noiseprofile_t dt_iop_denoiseprofile_get_auto_profile(dt_iop_module_t *self)
+static dt_noiseprofile_t dt_iop_denoiseprofile_get_auto_profile(dt_iop_module_t *self,
+                                                                GList *profiles,
+                                                                char *name,
+                                                                const size_t namelen,
+                                                                gboolean *autodetected,
+                                                                const gboolean compensate_hilite_pres)
 {
-  GList *profiles = dt_noiseprofile_get_matching(&self->dev->image_storage);
+  gboolean profiles_allocated = FALSE;
+  if(profiles == NULL)
+  {
+    profiles = dt_noiseprofile_get_matching(&self->dev->image_storage);
+    profiles_allocated = TRUE;
+  }
   dt_noiseprofile_t interpolated = dt_noiseprofile_generic; // default to generic poissonian
 
-  const int iso = self->dev->image_storage.exif_iso;
+  if(autodetected != NULL)
+  {
+    *autodetected = FALSE;
+  }
+  if(name != NULL)
+  {
+    g_strlcpy(name, _(interpolated.name), namelen);
+  }
+
+  const int exif_iso = self->dev->image_storage.exif_iso;
+  int iso = exif_iso;
+  int shift = 0;
+  if(compensate_hilite_pres)
+  {
+    const float hilight_pres = self->dev->image_storage.exif_highlight_preservation;
+    // Only compensate whole EV steps, as non-whole steps are (based on
+    // experience and discussion in #19624) handled by different means in the
+    // camera.
+    shift = floorf(hilight_pres);
+    if(shift >= 0)
+    {
+      iso >>= shift;
+    }
+    else
+    {
+      shift = 0;
+    }
+  }
   dt_noiseprofile_t *last = NULL;
   for(GList *iter = profiles; iter; iter = g_list_next(iter))
   {
     dt_noiseprofile_t *current = iter->data;
     if(current->iso == iso)
     {
       interpolated = *current;
+      if(autodetected != NULL)
+      {
+        *autodetected = TRUE;
+      }
+      if(name != NULL)
+      {
+        if(iso != exif_iso)
+        {
+          snprintf(name, namelen, _("found match for ISO %d (ISO %d %+d EV)"), iso, exif_iso, -shift);
+        }
+        else
+        {
+          snprintf(name, namelen, _("found match for ISO %d"), iso);
+        }
+      }
       break;
     }
     if(last && last->iso < iso && current->iso > iso)
     {
       interpolated.iso = iso;
       dt_noiseprofile_interpolate(last, current, &interpolated);
+      if(autodetected != NULL)
+      {
+        *autodetected = TRUE;
+      }
+      if(name != NULL)
+      {
+        if(iso != exif_iso)
+        {
+          snprintf(name, namelen, _("interpolated from ISO %d and %d (ISO %d %+d EV"), last->iso, current->iso,
```

Comment:
No way to make it shorter without making it obscure. Or maybe you have an idea?

Reply:
No, I've struggled with the wording, too.

The question is whether it's even necessary to give the user the detail from which profiles the interpolation was made. That seems more like a debugging detail to me. In the end, the user needs to know "is this an accurate profile or do I have to expect issues from interpolation maybe?".

If that's correct, then we could go with `"interpolated ISO %d (ISO %d %+d EV)", iso, exif_iso, -shift`.

Reply:
Works for me, indeed the final interpolated value is sufficient on the UI.